Tools/Algorithms/stencil.py

```diff
@@ -429,29 +433,77 @@ def run_main(dx, dy, dz, dt, nox, noy, noz, gamma=1., galilean=False, path='.',
 
 if __name__ == '__main__':
 
-    # --
-    # User can modify these input parameters
-    # --
-    # Cell size
-    dx = 1e-06
-    dy = 1e-06
-    dz = 2e-06
-    # Time step
-    dt = 1e-14
-    # Spectral order
-    nox = 8
-    noy = 8
-    noz = 16
-    # Lorentz boost
-    gamma = 30.
+    # Parse path to input file from command line
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--path')
+    args = parser.parse_args()
+    path = args.path
+
+    # Open input file
+    input_file = open(path, 'r').read()
```

Comment:
A suggestion here is to have a separate python script that reads in the input file into a dictionary (like the `parameters` dictionary in `yt`). Then all of the `re` processing wouldn't be needed. To get the number of cells would then be perhaps `nx = input_dict['amr.n_cell']` or `nx = [int(n) for n in  input_dict['amr.n_cell']]` depending on how much the script processes the input. That script would do something like this
```
for line in open(inputfilename):
    sline = line.split('=')
    parameters[sline[0].strip()] = sline[1].split()
```
(with extra checks to skip comments and blank lines and handle multiline parameters).

Reply:
Thank you for the suggestion, Dave! I have implemented it in [690d8aa](https://github.com/ECP-WarpX/WarpX/pull/4016/commits/690d8aa895db2f87f209305e5b231aa65e99a08e).
Just to confirm, do you want the parsing to be implemented in a separate Python file?

Reply:
I was thinking that it would be a generally useful thing to have a python script that could read in an input file. That's why I suggested putting it in a separate file.

Reply:
The function to parse the WarpX input is now called `parse_input` and defined in the separate file `parser.py`. This could be a good starting point for a more comprehensive WarpX input parser. For now, I would leave it in the same folder as the script `stencil.py`, since it is the only script using it.